bin/migrate-oats-data/applications/app_prep.py

```diff
@@ -314,22 +319,241 @@ def get_update_query_for_other():
     return query
 
 
-def map_oats_to_alcs_nfu_subtypes(oats_code):
-    # TODO this is work in progress and will be finished later
-    if oats_code == 1:
-        return "Accessory Buildings"
-    if oats_code == 2:
-        return "Additional Dwelling(s)"
-    if oats_code == 3:
-        return "Additional Structures for Farm Help"
-    if oats_code == 4:
-        return "Agricultural Land Use Remnant"
-    if oats_code == 5:
-        return "Agricultural Lease"
-    if oats_code == 6:
-        return "Agricultural Subdivision Remnant"
-
-    return ""
+def get_nfu_subtype_by_key(list_of_dicts, nfu_type_code, nfu_subtype_code):
+    for dict_obj in list_of_dicts:
+        if str(dict_obj["type_key"]) == str(nfu_type_code) and str(
+            dict_obj["subtype_key"]
+        ) == str(nfu_subtype_code):
+            return dict_obj["value"]
+
+    # Return None when no matching key found
+    return None
+
+
+def map_oats_to_alcs_nfu_subtypes(nfu_type_code, nfu_subtype_code):
+    nfu_subtypes = [
```

Comment:
Should this also be in a constants file somewhere?

Reply:
done